### ISSUE TITLE
Dependency version requirements may be excessively strict

### ISSUE THREAD

**Issue by spaddlewit** [pauldendulk/Mapsui#212]
I notice that in the NuGet packages, the version requirements for some of the dependencies might be unnecessarily strict.... for example:

SkiaSharp (>= 1.56.2 && < 1.57.0)
Newtonsoft.json (>= 9.0.0 && < 10.0.0)
BruTile (>= 0.21.1 && < 0.22.0)

I don't know about SkiaSharp or BruTile, but I am pretty certain that other versions of Newtonsoft.Json would work just fine with Mapsui.
Requiring exact versions of dependencies can cause issues when using Mapsui with other NuGet packages. If there is a bug in a dependency NuGet, it also prevents the user from upgrading that dependency without having to wait for Mapsui to publish an updated NuGet (such as in #209).

**Comment by pauldendulk** [comment]
This is complicated. The major version change signals a breaking change according to [semver](semver.org). Many project have breaking changes when upping the major version. I assume Newtonsoft.Json has too for some part of their public interface. From this perspective the current restriction is necessary. Updating to newer version could break the app. 

On the other hand updates to Newtonsoft.Json are not likely to break Mapsui. Many other projects have dependencies on Mapsui this could cause conflicts. In Mapsui Newtonsoft.Json

Some possible ways to go forward:
- Make the dependencies less restrictive as you suggested
- Turn the module that uses Newtonsoft.Json (ArcGis) into a separate nuget package, making conflicts less likely.
- Find some alternative for some simple json parsing. 
- Internalize Newtonsoft.Json: https://github.com/pauldendulk/Mapsui/issues/67

**Comment by Sebastian1989101** [comment]
Is it at least possible to upgrade the versions? In one of my projects I use Mapsui but also need functions from a newer version of SkiaSharp (currently I extract the function into a separate library and internalize SkiaSharp into it - but thats not very pretty).

**Comment by pauldendulk** [comment]
I have to read up on what is possible nowadays with nuget. What I would prefer is that I specify which version I compiled with. Right now I allow all patch versions. Actually also ones I did not test with, which is already a bit uncomfortable for me. Ideally it should be clear to the user what is the advised version, the version I tested with. 

To make it more flexible I could threat the minor version the same way, but this will mean that users will automatically get that version when they add Mapsui. This chance things break is higher there.

**Comment by pauldendulk** [comment]
Just curious, which function of the new skia are you using?

**Comment by Sebastian1989101** [comment]
Function was maybe not the correct term there. I currently work on my first AR app for the iPhone X. Mapsui runs fine as a 2D map inside of this environment already but if I try to render 2D-objects on a complete different location the "ObjectDisposedException" occurred from SkiaSharp. It seems to be fixed in the latest version.

**Comment by pauldendulk** [comment]
Ah, perhaps that was a re-fix on that issue you had earlier. I saw the issue was reopened and then fixed again. I hope to release soon.

But isn't it possible right now to override the nuspec? There is something like binding redirects.

**Comment by pauldendulk** [comment]
I made the dependencies a little bit more lenient

```
        <dependency id="SkiaSharp" version="[1.58.1,2.0)"/>
        <dependency id="Newtonsoft.json" version="[9.0,11.0)"/>
        <dependency id="BruTile" version="[0.21.1,0.23)"/>
```

- For skia I allow up to 2.0 (exclusive). This might actually help.
- For Newtonsoft I allow up to 11 (exclusive). I had to because I updated to 10 :). I think there should be at least be a limit on the major version.
- For BruTile it is still quite strict. After BruTile 1.0 that might become better.

I am closing this because I don't see any real good solution for this.
